pkg/fanal/walker/walk.go

```diff
@@ -91,5 +100,17 @@ func (w *walker) shouldSkipDir(dir string) bool {
 		}
 	}
 
+	if dir != "." && len(w.onlyDirs) > 0 {
+		for _, onlyDir := range w.onlyDirs {
```

Comment:
Does it work with the nested dir? Let's say `--only-dirs /*/bar/**` is passed. Then, Trivy walks from the root and reaches `/foo`. This function returns false as `/*/bar/**` doesn't match `/foo`. It never reaches `/foo/bar` even though `/foo/bar/baz.txt` exists. I may be missing something.

https://go.dev/play/p/wdZ1cZ89kSx

Reply:
You're correct. At first I didn't support wildcards but when rebasing, I saw the use of "doublestar.Match" and switched to it. It only works with leading "wildcards" - which brings no value - because of the case you mentionned.

I suggest toswitch back to use regular paths. WDYT ?

Reply:
I kept the patterns but apply them when considering files. When possible, it also skips some directories to avoid having traversing them.